components/badge.jsx

```diff
@@ -0,0 +1,62 @@
+var React       = require('react'),
+    ImageTag    = require('./imagetag.jsx');
+
+var Badge = React.createClass({
+    propTypes: {
+        title: React.PropTypes.string.isRequired,
+        status: React.PropTypes.string,
+        icon: React.PropTypes.string.isRequired,
+        icon2x: React.PropTypes.string.isRequired
+    },
+    render: function (){
+
+        /**
+         * returning the Status HTML
+         * @param statusVal
+         * @returns ReactElement
+         */
+        function status (statusVal) {
```

Comment:
try not to declare functions inside of `render`, because that rebuilds them on every render call. Instead, we can define them as part of the Badge class prototype:
```
var Badge = React.createClass({
  proptTypes: ...,
  status: function(statusVal) {
    ...
  },
  render: function() {
    ...
      statusElement  = this.status(...);
    ...
  }
...
});
```